sdks/python/setup.py

```diff
@@ -124,7 +124,7 @@ def get_version():
     'pytz>=2018.3,<=2018.4',
     'pyyaml>=3.12,<4.0.0',
     'pyvcf>=0.6.8,<0.7.0',
-    'typing>=3.6.0,<3.7.0',
+    'typing>=3.6.0,<3.7.0; python_version < "3.7.0"',
```

Comment:
I was not aware of this usage. If this works, we can probably consolidate our  REQUIRED_PACKAGES_PY3_ONLY etc parts here as well.

(Similarly we can use a single requirements base file for 2 different Docker files for py2 and py3)

@tvalentyn

Reply:
Thanks for your change, @manuzhang, I was also not aware of this syntax. It is documented in https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies and in more detail in https://www.python.org/dev/peps/pep-0508/. We should use it to simplify our setup.py and instead of `REQUIRED_PACKAGES_PY3_ONLY`/`REQUIRED_PACKAGES_PY2_ONLY`. @manuzhang - would you be interested in adding this cleanup to this PR?

As for this change: 
- I do notice that more typehints tests fail for Python 3.7, but I did not see any effect whether we install `typing` or not. I did see errors coming from `futures` package, and removing it helped to significantly reduce the number of failures on 3.7. As per https://pypi.org/project/typing/: in Python 3.5 and later, the typing module lives in the stdlib, and installing this package has no effect. I think we should file a Jira and do a little more investigation to rootcause the issue you are facing. I wonder if it may be platform-specific.

Reply:
@aaltay @tvalentyn I'd love to make the cleanup in a separate PR. I've seen errors coming from `futures` package from Python 3.5 to 3.7 and interestingly *installing futures* solves it for me. I've logged the typing issue in [BEAM-5607](https://issues.apache.org/jira/browse/BEAM-5607) and feel free to add more info there. By the way, do you think it's worthwhile to set up a testing matrix for Python 2, 3.5, 3.6 and 3.7 on Jenkins respectively ?

Reply:
Thanks, Manu, I'll come back to BEAM-5607. It's definitely worthwhile to also add tox suites for all 3.4 - 3.7. Hopefully in a way we can run the suites for different versions in parallel. Feel free to add a feature request as subtask for BEAM-1251 to add test coverage. 

Given `typing` description on PyPi, should we set `python_version < "3.5.0"?